#1052 Not check for deprecation messages in special case

xupyprmv opened the pull request:
#1052 `StartsDaemonITCase` class, `deprecatesDefaultImage` method. Avoid checking deprecation messages when the repo is the actual Rultor repo: https://github.com/yegor256/rultor . Also I removed `@Ignore` annotation from test.

alex-palevsky commented:
@xupyprmv Let me find a reviewer for this pull request, thanks for submitting it

alex-palevsky commented:
@mkordas could you review this pull request

mkordas commented:
@xupyprmv on it

mkordas commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -115,27 +115,31 @@ public void startsDaemon() throws Exception {
-     *  https://github.com/yegor256/rultor
      */
     @Test
     @Ignore
```
@xupyprmv what is the puzzle to unignore this test?

xupyprmv replied:
I think there is no puzzle for it. Do I have to add a puzzle?

mkordas replied:
@xupyprmv let's ask our architect about it

mkordas replied:
@original-brownbear apparently we have ignored test and there's no puzzle to unignore it. What do you suggest here?

original-brownbear replied:
@mkordas @xupyprmv Sorry this was badly reviewed by me in terms of the wording of the puzzle.
This puzzle should have included unignoring the test. Please unignore the test here.

mkordas commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -115,27 +115,31 @@ public void startsDaemon() throws Exception {
+        )) {
+            for (final String path
+                : xml.xpath("/p/entry[@key='merge']/entry[@key='script']")
+                 ) {
```
@xupyprmv I think that formatting here is a bit broken, can you reformat the code using IDEA?

mkordas commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -115,27 +115,31 @@ public void startsDaemon() throws Exception {
-                );
+        if (!"yegor256/rultor".equals(
+            xml.xpath("/wire/github-repo/text()").get(0)
+        )) {
```
@xupyprmv we need to apply this here: http://www.yegor256.com/2014/10/23/paired-brackets-notation.html

mkordas commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -115,27 +115,31 @@ public void startsDaemon() throws Exception {
-                    dir,
-                    StringStartsWith.startsWith("#### Deprecation Notice ####")
-                );
+        if (!"yegor256/rultor".equals(
```
@xupyprmv we have duplicated literal now, right? Why Qulice is not complaining?

mkordas commented:
@xupyprmv please see comments

xupyprmv commented:
@mkordas Thanks for CR. Please continue.

mkordas commented:
@xupyprmv looks good to me, except puzzle concern

mkordas commented:
@rultor merge

rultor commented:
> @rultor merge

@mkordas Thanks for your request. @original-brownbear Please confirm this.

original-brownbear commented:
@xupyprmv could you please unignore the test here ?

xupyprmv commented:
@original-brownbear Sure, I updated my PR. I also added one more condition, to let test pass (looks like `/wire/github-repo` is not mandatory). Please continue.

mkordas commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -115,27 +120,31 @@ public void startsDaemon() throws Exception {
-                    dir,
-                    StringStartsWith.startsWith("#### Deprecation Notice ####")
-                );
+        final List<String> repo = xml.xpath("/wire/github-repo/text()");
```
@xupyprmv I'd inline `.get(0)` here, as now these are rather `repos`

xupyprmv replied:
@mkordas I changed name to `repos`. Inline will not have good effect here, because it leads to double execution and extracting path to constant.

mkordas replied:
@xupyprmv looks good!

original-brownbear commented:
@xupyprmv could you please add a clear description of your changes to the PR description. Looks good btw, proper description is needed though :)

xupyprmv commented:
@original-brownbear Description was fixed.

original-brownbear commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -70,6 +70,11 @@
     /**
+     * Rultor repo.
+     */
+    private static final String RULTOR = "yegor256/rultor";
```
@xupyprmv this can just be a local variable in `deprecatesDefaultImage`, it's only used there :) No need to add this kind of noise. Putting this kind of thing outside of the function and into a constant only makes sense if we really need Javadoc on it, but I think here we're both clear on the meaning of the variable :)

mkordas replied:
@original-brownbear `deprecatesDefaultImage` is not a legal name for local variable if Qulice is used

original-brownbear commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -115,27 +120,33 @@ public void startsDaemon() throws Exception {
-                    StringStartsWith.startsWith("#### Deprecation Notice ####")
-                );
+        final List<String> repos = xml.xpath("/wire/github-repo/text()");
+        if (
```
@xupyprmv just for my peace of mind :) This looks just fine to me, but I'm slightly fearful someone might break `this.talk()` inadvertently ...
Can we just adjust this in a trivial manner please ? :)

* move  `final String dir` in front of the first `if` ( l.131 currently)
* Move this section at the end of the function
```java
                    MatcherAssert.assertThat(
                         dir,
                         StringStartsWith.startsWith(
                             "#### Deprecation Notice ####"
                         )
                     );
```

This way this will always and forever work and also is easier to read imo :)

xupyprmv replied:
@original-brownbear Really good advice, thanks. Now I understand why previous version was bad.  Eagle eye!

original-brownbear commented:
@xupyprmv sorry missed two spots last night, my bad.
But both are fully trivial things, please address those real quick.
Thanks!

xupyprmv commented:
@original-brownbear I updated PR. Please continue.

original-brownbear commented on src/test/java/com/rultor/agents/daemons/StartsDaemonITCase.java:
```diff
@@ -126,28 +121,20 @@ public void startsDaemon() throws Exception {
+            MatcherAssert.assertThat(
+                dir,
+                StringStartsWith.startsWith("#### Deprecation Notice ####")
+            );
```
@xupyprmv this doesn't really change anything the matcher is still wrapped in an if :) If the if evaluates to `false` this test passes without testing anything. The matching needs to be outside the if.

xupyprmv replied:
> his doesn't really change anything

@original-brownbear There was a bug with double reading of stream. P.S. I will fix matcher soon.

xupyprmv commented:
@original-brownbear I updated PR according your note about `Matcher`. Please continue.

original-brownbear commented:
@rultor merge please :)

original-brownbear commented:
@xupyprmv thanks!

rultor commented:
> @rultor merge please :)

@original-brownbear OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/7920-204101684)

rultor commented:
> @rultor merge please :)

@original-brownbear Done! FYI, the full log is [here](http://www.rultor.com/t/7920-204101684) (took me 18min)

original-brownbear commented:
@rultor deploy please :)

rultor commented:
> @rultor deploy please :)

@original-brownbear OK, I'll try to deploy now. You can check the progress [here](http://www.rultor.com/t/7920-204107136)

rultor commented:
> @rultor deploy please :)

@original-brownbear Done! FYI, the full log is [here](http://www.rultor.com/t/7920-204107136) (took me 10min)

original-brownbear commented:
@xupyprmv please close this PR, since it was a rebase it wasn't automatically closed :)

original-brownbear commented:
@alex-palevsky please pay attention here, this has been closed for 3 days now.

alex-palevsky commented:
@mkordas thanks, I just added **28 mins** to your account, payment `5701103ca11a1f68f8000219`, 47 hours and 56 mins spent; extra minutes for review comments (c=13); +28 to your rating, your total score is [+9252](http://www.netbout.com/b/36080?open=rating)
